src/sobol_sensitivity.jl

```diff
@@ -342,10 +342,18 @@ function gsa_sobol_all_y_analysis(method, all_y::AbstractArray{T}, d, n, Ei_esti
         nboot > 1 ? reshape(ST_CI, size_...) : nothing)
 end
 
-function gsa(f, method::Sobol, p_range::AbstractVector; samples, kwargs...)
+function gsa(f, method::Sobol, p_range::AbstractVector; samples,
+        rng::AbstractRNG = Random.default_rng(), kwargs...)
+    log2num = ceil(Int, log2(samples))
+    samples2n = 2^log2num
+    if samples2n != samples
+        samples = samples2n
+        @warn "Passed samples is not a power of 2, number of sample points changed to $samples"
+    end
```

Comment:
Alternatively, one could throw an error if `samples` is not a power of 2 with e.g.
```suggestion
    log2num = QuasiMonteCarlo.logi(2, samples)
```
One should really not use non-2^x subsets of a Sobol sequence, see e.g. https://github.com/JuliaMath/Sobol.jl/issues/31.

Reply:
If samples are increased to a power of 2 automatically, I think one could make use of `nextpow` (which takes care of rounding issues etc.):
```suggestion
    samples2n = nextpow(2, samples)
    if samples2n != samples
        samples = samples2n
        @warn "Passed samples is not a power of 2, number of sample points changed to $samples"
    end
    log2num = round(Int, log2(samples))
```

Reply:
I have seen that discussion - I don't want to throw an error because that would be breaking change and I don't want to do that . Thanks for the nextpow suggestion, I had a feeling I should look up a function to this 😄